File: modify.py
def modify(a, n):
	if n<=1:
		return

	prev = a[0]
	a[0] = a[0] * a[1]

	for i in range(1,n-1):
		curr = a[i]

		a[i] = prev * a[i+1]

		prev = curr

	a[n-1] = prev * a[n-1]


def print_list(a):
	for i in range(0, len(a)):
		print(a[i])

a = [2,3,4,5,6]
modify(a, len(a))
print("modified list is")
print_list(a)

File: basic_prob/modify.cpp
#include<iostream>
using namespace std;

void modify(int a[], int n)
{
	if(n <=	 1)
		return;
	int prev = a[0];
	a[0] = a[0] * a[1];
	for(int i = 1; i<n-1; i++)
	{
		int curr = a[i];

		a[i] = prev * a[i+1];

		prev = curr;
	}
	a[n-1] = prev * a[n-1];
}

void print_array(int a[], int n)
{
	for(int i=0; i<n; i++)
	{
		cout<<a[i]<<" ";
	}
}

int main()
{
	int a[] = {2, 3, 4, 5, 6};
	int n = sizeof(a)/sizeof(a[0]);
	modify(a, n);
	cout<<"modified array is"<<" ";
	print_array(a,n);
	return 0;
}